lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -858,10 +859,12 @@ namespace Js
 
         for (uint i = 0; i < ownPropsArray->GetLength(); i++)
         {
-            ownPropsArray->DirectGetItemAt(i, &propKey);
+            BOOL getPropResult = ownPropsArray->DirectGetItemAt(i, &propKey);
+            Assert(getPropResult);
```

Comment:
Assuming the object isn't a `Proxy`, `JavascriptOperators::GetOwnPropertyKeys` just returns the result of `JavascriptObject::CreateKeysHelper`, which is always a `JavascriptArray`.

I understood @Yongqu to be concerned about catching a change in something along the chain of calls after `JavascriptOperators::GetOwnPropertyKeys` that such that it *doesn't* ultimately return an Array (hence an assert being appropriate here). I just noticed that `JavascriptArray::FromVar` would already catch such a case, though, since `FromVar` includes the assert:
`AssertMsg(Is(aValue), "Ensure var is actually a 'JavascriptArray'");`

So the one I added is redundant, it looks like. 

Both `JavascriptConversion::ToPropertyKey` and `ownPropsArray->DirectGetItemAt` return a BOOL indicating success/failure, although AFAICT the implementation of `JavascriptConversion::ToPropertyKey` *always* succeeds. That doesn't mean it always will, though, and `JavascriptObject::GetValuesOrEntries` currently includes an Assert identical to the one I added (see https://github.com/Microsoft/ChakraCore/blob/master/lib/Runtime/Library/JavascriptObject.cpp#L1202)